backend/account/business.py

```diff
@@ -5,11 +5,72 @@
 """
 
 from . import query
-from .models import User
+from .models import *
 
 def get_or_create_user(payload: dict) -> User:
-    return query.get_or_create_user(payload)
+    is_new = not query.user_exists(payload['google_id'])
+    user = query.get_or_create_user(payload)
+    if (is_new):
+        create_default_columns(payload['google_id'])
+
+    return user
 
 
 def update_user(payload: dict):
     query.update_user(payload)
+
+
+def get_columns(credential: str) -> list[KanbanColumn]:
```

Comment:
Or maybe we're using the id field of the user table to identify users? (other than on login attempt which will use the google_id (sub) field?) I might need to review the model, but either way i don't think credential is the right word 🤔

Reply:
Django automatically adds a unique numerical `id` field to any model that doesn't specify a primary key (more info [here](https://docs.djangoproject.com/en/4.1/topics/db/models/#automatic-primary-key-fields)), so the `User` model's primary key is `id`, rather than `google_id`.

I agree that `google_id` is a better name for that parameter, but it might be better to make it use the user's `id` instead... or maybe make it so `google_id` is the user's primary key? Not sure. I think it might be better to use `google_id` as the primary key for the sake of having one unique identifier instead of two, but I don't know if it's bad practice to be using the `google_id` for things other than logging in. Any thoughts?

Reply:
I don't feel strongly about either way - might be best to just use `id` for everything other than `get_or_create_account` yeah

Reply:
Makes sense, I'll change `get_columns` to use `id` (and see if there are other places I could make a similar change) later today. I think I'll keep the name `id` though, to make it clear that we're using the Django default primary key.

Reply:
Not super invested in naming, but I feel like `id` is too generic, **because** every Django model has that field and we're not specifying what the id is referring to - I'll send a message and see what others think